src/Mobs/Path.cpp

```diff
@@ -213,6 +207,76 @@ bool cPath::Step_Internal()
 
 
 
+void cPath::AttemptToFindAlternative()
+{
+	Vector3i PotentialNewDestination;
+	// Attempt to find a nearby path.
+	int OriginalX = (rand() % 5) - 2;
+	int OriginalZ = (rand() % 5) - 2;  // A random value between -2 and 2.
+	int x, y, z;
+	x = OriginalX;
+	z = OriginalZ;
+	// Scan a 5x3x5 cube around the original path, starting with a random spot.
+	// TODO: scan a 5x1x5 and If a block in the scan area is air, we scan the solid below instead.
+	for (;;)
+	{
+		for (;;)
```

Comment:
I think making this `for(int z = OriginalZ; z != OriginalZ; z = (z == 3 ? -2 : z + 1))` would be clearer